Spanner/src/TransactionalReadTrait.php

```diff
@@ -267,14 +267,19 @@ public function execute($sql, array $options = [])
         $this->singleUseState();
         $this->checkReadContext();
 
-        $options['transactionId'] = $this->transactionId;
+        if (is_null($this->transactionId) && isset($this->options['begin'])) {
```

Comment:
It makes me nervous to put this here in a trait because `$this->options` isn't defined in this trait. If this is a requirement of this trait, we should define it in this trait (rather than in `Transaction` and `SnapshotTrait`).

Reply:
The `$options` is only defined in the this `TransactionalReadTrait`. See here: https://github.com/googleapis/google-cloud-php/blob/d1a5c3832f521891b82f75346c5d395d88f058c0/Spanner/src/TransactionalReadTrait.php#L63